quinn-proto/src/crypto/ring.rs

```diff
@@ -70,3 +70,72 @@ impl crypto::HmacKey for hmac::Key {
         hmac::verify(self, data, signature).map_err(|_| ())
     }
 }
+
+fn make_key<K: aead::BoundKey<ZeroNonceSequence>>(
+    algorithm: &'static aead::Algorithm,
+    key: &[u8],
+) -> K {
+    let key = aead::UnboundKey::new(algorithm, key).unwrap();
+    K::new(key, ZeroNonceSequence::new())
+}
+
+pub struct Aes256GcmKey {
+    opening_key: aead::OpeningKey<ZeroNonceSequence>,
+    sealing_key: aead::SealingKey<ZeroNonceSequence>,
+}
```

Comment:
This can be replaced entirely with *ring*'s `LessSafeKey`. That'll avoid duplicating storage and make the `ZeroNonceSequence` helper unnecessary. Further, `open` and `seal` will then be able to take `&self` rather than `&mut self`.